### PR TITLE
Custom AWS CF templates

### PR DESCRIPTION
**Pull request by joel-hamill**
How to generate custom AWS templates from installer.

### CODE REVIEW

**Comment by pyronicide** [comment]
@joel-hamill let's get someone to user test this.

File: 1.8/administration/installing/cloud/aws/advanced/aws-custom.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+* An Amazon S3 bucket with read-write access.
+
+
+1.  Download the [dcos_generate_config.sh](https://dcos.io/releases/) to your bootstrap node.
+1.  Create a directory named `genconf` on your node and navigate to it.
+    
+    ```bash
+    $ mkdir -p genconf
+    ```
+1.  Create a configuration file and save as `config.yaml`. This configuration file specifies your AWS credentials and the S3 location to store the generated artifacts. These are the required parameters:
```
Comment (joel-hamill) [inline 1.8/administration/installing/cloud/aws/advanced/aws-custom.md]: Change to `From your `genconf` directory, create a configuration...`

Comment (surdy) [inline 1.8/administration/installing/cloud/aws/advanced/aws-custom.md]: Make sure to let the users know that config.yaml needs to be generated in genconf directory

File: 1.8/administration/installing/cloud/aws/advanced/aws-custom.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    $ sudo bash dcos_generate_config.sh --aws-cloudformation
+    ```
+
+     The root URL for this bucket location is printed at the end of this step. You should see a message like this:
+    
+    ```bash
+    AWS CloudFormation templates now available at: https://<amazon-web-endpoint>/<path-to-directory>
+    ```
+1.  Go to [CloudFormation](https://console.aws.amazon.com/cloudformation/home) and click **Create Stack**.
+1.  On the **Select Template** page, specify the Amazon S3 template URL path to your Zen template. For example, `https://s3-us-west-2.amazonaws.com/templates/dcos/cloudformation/el7-zen-1.json`.
```
Comment (surdy) [inline 1.8/administration/installing/cloud/aws/advanced/aws-custom.md]: For reference I used s3 bucket `harpreet-test` region `us-west-2` and bucket path `templates` the resulting URL for el7-zen-1.json looked like `https://s3-us-west-2.amazonaws.com/harpreet-test/templates/config_id/14221b9104081387447be59e178458749d654c7f/cloudformation/el7-zen-1.json`

File: 1.8/administration/installing/cloud/aws/advanced/aws-custom.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+**Prerequisites:**
+
+* A node that meets custom installer bootstrap node [system requirements](/docs/1.8/administration/installing/custom/system-requirements/).
+* Advanced template [system requirements](/docs/1.8/administration/installing/cloud/aws/advanced/system-requirements/).
+* An Amazon S3 bucket with read-write access.
+
+
+1.  Download the [dcos_generate_config.sh](https://dcos.io/releases/) to your bootstrap node.
+1.  Create a directory named `genconf` on your node and navigate to it.
```
Comment (sascala) [inline 1.8/administration/installing/cloud/aws/advanced/aws-custom.md]: should they do this from their home directory, or from anywhere?

File: 1.8/administration/installing/cloud/aws/advanced/aws-custom.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    **aws_template_upload**
+    Specify whether to automatically upload the customized advanced templates to your S3 bucket. For example, `aws_template_upload: true`. If you specify `true`, you must also specify these parameters:
+    
+    *  **aws_template_storage_access_key_id**
+       Specify the AWS [Access Key ID](http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSGettingStartedGuide/AWSCredentials.html). For example, `aws_template_storage_access_key_id: AKIAIOSFODNN7EXAMPLE`  
+        
+    *  **aws_template_storage_secret_access_key**
+       Specify the AWS [Secret Access Key](http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSGettingStartedGuide/AWSCredentials.html). For example, `aws_template_storage_secret_access_key: wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY`.
+    
+1.  Run the DC/OS installer script with the AWS argument specified. This command creates and uploads a custom build of the DC/OS artifacts and templates uploaded to the specified S3 bucket.
```
Comment (sascala) [inline 1.8/administration/installing/cloud/aws/advanced/aws-custom.md]: is "uploaded" necessary?

File: 1.8/administration/installing/cloud/aws/advanced/aws-custom.md (hunk 3 above)
Comment (joel-hamill) [inline 1.8/administration/installing/cloud/aws/advanced/aws-custom.md]: should be home

Comment (sascala) [inline 1.8/administration/installing/cloud/aws/advanced/aws-custom.md]: So, "create a directory named `genconf` in your home directory"

File: 1.8/administration/installing/cloud/aws/advanced/aws-custom.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+* A node that meets custom installer bootstrap node [system requirements](/docs/1.8/administration/installing/custom/system-requirements/).
+* Advanced template [system requirements](/docs/1.8/administration/installing/cloud/aws/advanced/system-requirements/).
+* An Amazon S3 bucket with read-write access.
+
+
+1.  Download the [dcos_generate_config.sh](https://dcos.io/releases/) to your bootstrap node.
+1.  Create a directory named `genconf` on your node and navigate to it.
+    
+    ```bash
+    $ mkdir -p genconf
```
Comment (sascala) [inline 1.8/administration/installing/cloud/aws/advanced/aws-custom.md]: $ cd $HOME
$ mkdir -p genconf

File: 1.8/administration/installing/cloud/aws/advanced/aws-custom.md (hunk 4 above)
Comment (joel-hamill) [inline 1.8/administration/installing/cloud/aws/advanced/aws-custom.md]: Changed to `This command creates and uploads a custom build of the DC/OS artifacts and templates to the specified S3 bucket.`

**Pull request merged**